### PR TITLE
invalidate empty index name and add unit test

### PR DESCRIPTION
**Pull request by xinyiyelp**
The server still tries to launch an index when its name is empty, which was never created before. 

The work here is to invalidate empty index name before everything starts.

### CODE REVIEW

File: src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java [hunk 1]
```diff
@@ -716,6 +716,13 @@ public void settingsV2(
     public void startIndex(
         StartIndexRequest startIndexRequest, StreamObserver<StartIndexResponse> responseObserver) {
       logger.info("Received start index request: {}", startIndexRequest);
+      if (startIndexRequest.getIndexName().isEmpty()) {
+        responseObserver.onError(
```
Comment (swekannan) [inline src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java:720]: You can add a logger.warn() before this error handling to add info on the indexName for which this failed. And make the error more descriptive such as "error while trying to start index since indexName was empty: " + indexName

File: src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java [hunk 2]
```diff
@@ -277,6 +277,24 @@ public void testStartShard() throws IOException {
...
 
+  @Test
+  public void testStartIndex() {
+    String testIndex = grpcServer.getTestIndex();
+    LuceneServerGrpc.LuceneServerBlockingStub blockingStub = grpcServer.getBlockingStub();
+    // create the index
+    blockingStub.createIndex(CreateIndexRequest.newBuilder().setIndexName(testIndex).build());
+    try {
+      // start the index
+      String emptyTestIndex = "";
```
Comment (swekannan) [inline src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java:288]: Also test cases where the indexName is not set in addition to testing for empty strings.

File: src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java (hunk 1 above)
Comment (xinyiyelp) [reply src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java:721]: Yup, added warning with descriptive message

File: src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java (hunk 2 above)
Comment (xinyiyelp) [reply src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java:299]: Added another test case: launch the index with empty string before creating a index.

Comment (sarthakn7) [reply src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java:299]: Can we add these two different test cases in two different functions? The create index test can be moved into a test called `testCreateIndexWithEmptyString`

File: src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java (hunk 1 above)
Comment (sarthakn7) [reply src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java:721]: Empty index name in an error might be confusing (and it's always going to be ""). Maybe skip the index name or log the start index request?

Comment (swekannan) [reply src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java:721]: Logging the startIndex request sounds like a better option. Will it show when the indexName string is not set vs an empty string since it seems like protobuf sets string defaults to ""?

Comment (sarthakn7) [reply src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java:721]: It doesn't show unset only empty unfortunately

Comment (xinyiyelp) [reply src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java:721]: Thank you for reviewing this! @sarthakn7 @swekannan 
Since line 718 has already logged out the `startIndexRequest`, here I skipped the index name and warned it's an empty index name.

File: src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java (hunk 2 above)
Comment (xinyiyelp) [reply src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java:299]: It sounds a good idea, I splitted the unit tests into two.
For the naming, I name it more specifically: `testCreateAndStartIndexWithEmptyString` since we are actually testing the start index functionality.

Comment (sarthakn7) [reply src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java:304]: Ohh my bad @swekannan I thought we were testing the createIndex as well. Yeah just test for startIndex is fine if we have not changed anything in the create.

File: src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java [hunk 3]
```diff
@@ -277,6 +277,41 @@ public void testStartShard() throws IOException {
...
+    } catch (StatusRuntimeException e) {
+      assertEquals(
+          String.format(
+              "INVALID_ARGUMENT: error while trying to start index since indexName was empty: "),
+          e.getMessage());
+    }
+  }
+
+  @Test
+  public void testCreateAndStartIndexWithEmptyString() {
```
Comment (swekannan) [inline src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java:297]: Hi @xinyiyelp you can remove this test then and only keep the startIndex test. You can then re-request review for approval.

**Review by swekannan — APPROVED** [review]
